symphony/lib/toolkit/class.datasource.php

```diff
@@ -536,6 +536,38 @@ public static function removeEscapedCommas($string)
         return preg_replace('/(?<!\\\\)\\\\,/', ',', $string);
     }
 
+    private static $digitReplacements = array(
+        '1' => 'one',
+        '2' => 'two',
+        '3' => 'three',
+        '4' => 'four',
+        '5' => 'five',
+        '6' => 'six',
+        '7' => 'seven',
+        '8' => 'eight',
+        '9' => 'nine',
+        '0' => 'zero',
+    );
+    /**
+     * XML does not allow tags or attribute to begin with a number.
+     * This function will replace the leading digit with its english equivalent.
+     *
+     * @param string $value
+     * @return string
+     *  The escaped name
+     */
+    public function escapeFirstCharIfDigit($value)
+    {
+        foreach (static::$digitReplacements as $d => $r) {
+            if (strpos($value, $d) === 0) {
+                $value = substr($value, 1);
+                $value = lcfirst($value);
```

Comment:
Why should this function deal with upper/lower case letters? That's literally "out if scope" IMHO.

Reply:
Hum I think this is an error.